### ISSUE TITLE
[Feature] Development guide

### ISSUE THREAD

**Issue by mmontes11** [mariadb-operator/mariadb-operator#144]
Create a development guide to make new contributor's life easier. For example:
- https://github.com/external-secrets/external-secrets/blob/main/docs/contributing/devguide.md

It should include:
- Installation requirements
- Cluster setup
- Run `mariadb-operator` locally
- Test locally
- Local networking setup

Some bits have already been discussed in our slack channel:
- https://mariadb-operator.slack.com/archives/C056RAECH0W/p1684316819408259?thread_ts=1684231386.023899&cid=C056RAECH0W
- https://mariadb-operator.slack.com/archives/C056RAECH0W/p1687522077156369

In summary, to get a dev environment up and running we need (see [CI](https://github.com/mariadb-operator/mariadb-operator/blob/main/.github/workflows/ci.yml) for reference):
```bash
make cluster
make install
make net
make run
```
